src/sinks/mqtt/config.rs

```diff
@@ -0,0 +1,212 @@
+use std::time::Duration;
+
+use rand::Rng;
+use rumqttc::{MqttOptions, QoS, TlsConfiguration, Transport};
+use snafu::{ResultExt, Snafu};
+use vector_lib::codecs::JsonSerializerConfig;
+
+use crate::template::Template;
+use crate::{
+    codecs::EncodingConfig,
+    config::{AcknowledgementsConfig, Input, SinkConfig, SinkContext},
+    sinks::{
+        mqtt::sink::{ConfigurationSnafu, MqttConnector, MqttError, MqttSink, TlsSnafu},
+        prelude::*,
+        Healthcheck, VectorSink,
+    },
+    tls::{MaybeTlsSettings, TlsEnableableConfig},
+};
+
+/// Configuration for the `mqtt` sink
+#[configurable_component(sink("mqtt"))]
+#[derive(Clone, Debug)]
+pub struct MqttSinkConfig {
+    /// MQTT server address (The broker’s domain name or IP address).
+    #[configurable(metadata(docs::examples = "mqtt.example.com", docs::examples = "127.0.0.1"))]
+    pub host: String,
+
+    /// TCP port of the MQTT server to connect to.
+    #[serde(default = "default_port")]
+    pub port: u16,
+
+    /// MQTT username.
+    pub user: Option<String>,
+
+    /// MQTT password.
+    pub password: Option<String>,
+
+    /// MQTT client ID.
+    pub client_id: Option<String>,
+
+    /// Connection keep-alive interval.
+    #[serde(default = "default_keep_alive")]
+    pub keep_alive: u16,
+
+    /// If set to true, the MQTT session is cleaned on login.
+    #[serde(default = "default_clean_session")]
+    pub clean_session: bool,
+
+    #[configurable(derived)]
+    pub tls: Option<TlsEnableableConfig>,
+
+    /// MQTT publish topic (templates allowed)
+    pub topic: Template,
+
+    #[configurable(derived)]
+    pub encoding: EncodingConfig,
+
+    #[configurable(derived)]
+    #[serde(
+        default,
+        deserialize_with = "crate::serde::bool_or_struct",
+        skip_serializing_if = "crate::serde::is_default"
+    )]
+    pub acknowledgements: AcknowledgementsConfig,
+
+    #[configurable(derived)]
+    pub quality_of_service: MqttQoS,
+}
+
+/// Supported Quality of Service types for MQTT.
+#[configurable_component]
+#[derive(Clone, Copy, Debug, Derivative)]
+#[derivative(Default)]
+#[serde(rename_all = "lowercase")]
+#[allow(clippy::enum_variant_names)]
+pub enum MqttQoS {
+    /// AtLeastOnce.
+    #[derivative(Default)]
+    AtLeastOnce,
+
+    /// AtMostOnce.
+    AtMostOnce,
+
+    /// ExactlyOnce.
+    ExactlyOnce,
```

Comment:
I noticed you changed the derivative default here. And I noticed in the rendered cue files we seem to not have a default, meaning the `quality_of_service` is a required field.  I'm wondering if the cue files need to be re-generated.

You can try `make check-component-docs`. to see if that is the case. If you don't already have CUE installed, this might not work for you. LMK and I can run this for you.

Reply:
You're right. I've added a default function and now it is regenerated with the correct default.